File: problem_set/c/addcomma3.c
#include <stdio.h>
#include <string.h>

int main (){
  char buffer[50], answer[50] = "";  
  long remainder, number;
  int count = 0;
  scanf("%d", &number);
  while(number > 0){
	remainder = number % 10;
	number = number / 10;
	count++;
	strcpy(buffer, answer);
	sprintf (answer, "%d%s", remainder, buffer);	
	if(count % 3 == 0 && number != 0){
		strcpy(buffer, answer);
		sprintf (answer, ",%s", buffer);
	}
  }
  printf ("%s\n", answer);
  return 0;
}

File: problem_set/python/addcomma3.py
'''PROGRAM : Separate Comma
   WRITER  : NAT KANJANASIRI
   DATE    : JAN 30, 2023
   PURPOSE : '''
number = int(input())
answer = ''
while number > 0:
    remainder = number % 1000
    number = number // 1000
    if number > 0:        
        if remainder == 0:
            answer = ',' + '000' + answer
        else:
            answer = ',' + str(remainder) + answer
    else:
        answer = str(remainder) + answer
print(answer)